atomic_reactor/plugins/post_export_operator_manifests.py

```diff
@@ -0,0 +1,139 @@
+        Check if the plugin is running in orchestrator.
+
+        :return: bool
+        """
+        if get_platforms(self.workflow):
+            return True
+        return False
+
+    def should_run(self):
+        """
+        Check if the plugin should run or skip execution.
+
+        :return: bool, False if plugin should skip execution
+        """
+        if self.is_orchestrator():
+            self.log.warning("%s plugin set to run on orchestrator. Skipping", self.key)
+            return False
+        if self.operator_manifests_extract_platform != self.platform:
+            self.log.info("Only platform [%s] will upload operators metadata. Skipping",
+                          self.operator_manifests_extract_platform)
+            return False
+        if is_scratch_build():
+            self.log.info("Scratch build. Skipping")
+            return False
+        if not self.has_operator_manifest():
+            self.log.info("Operator manifests label not set in Dockerfile. Skipping")
+            return False
+        return True
+
+    def run(self):
+        """
+        Run the plugin.
+
+        This plugin extracts the operator manifest files from an image, saves
+        them as a zip archive, and returns its path
+
+        :return: str, path to operator manifests zip file
+        """
+        if not self.should_run():
+            return
+
+        manifests_archive_dir = tempfile.mkdtemp()
+        image = self.workflow.image
+        # As in flatpak_create_oci, we specify command to prevent possible docker daemon errors.
+        container_dict = self.tasker.d.create_container(image, command=['/bin/bash'])
+        container_id = container_dict['Id']
+        try:
+            bits, stat = self.tasker.d.get_archive(container_id,
+                                                   os.path.join('/', MANIFESTS_DIR_NAME))
+        except APIError as ex:
+            self.log.error('Could not extract operator manifest files: %s', ex)
+            raise RuntimeError('Could not extract operator manifest files: %s' % ex)
+        finally:
+            self.tasker.d.remove_container(container_id)
+
+        fd, extracted_file = tempfile.mkstemp()
+        os.close(fd)
+        with open(extracted_file, 'wb') as f:
+            for chunk in bits:
+                f.write(chunk)
+
+        tar_archive = tarfile.TarFile(extracted_file)
+        tar_archive.extractall(manifests_archive_dir)
+        manifests_path = os.path.join(manifests_archive_dir, MANIFESTS_DIR_NAME)
+        base_zipfile_name = os.path.join(manifests_archive_dir, OPERATORS_ARCHIVE_NAME)
+        manifest_files = os.listdir(manifests_path)
+        if not manifest_files:
+            self.log.error('Empty operator manifests directory')
+            raise RuntimeError('Empty operator manifests directory')
+
+        self.log.debug("Archiving operator manifests: %s", manifest_files)
+        manifests_zipfile_path = shutil.make_archive(base_zipfile_name,
```

Comment:
There is unexpected dir named `.` in the archive
```
$ unzip -l operator_manifests.zip 
Archive:  operator_manifests.zip
  Length      Date    Time    Name
---------  ---------- -----   ----
        0  03-05-2019 08:10   ./
        4  03-05-2019 08:10   empty.yaml
---------                     -------
        4                     2 files
```

Reply:
This is was a [known bug in shutil](https://bugs.python.org/issue28488) which [has been fixed](https://github.com/python/cpython/commit/666de7772708047b63125126b0147931571254a4) in python >= 3.5 and in newer python 2.7 releases.

It is present in python <= 2.7.11 though, which is the version we got in CentOS 7. I will include some code to deal with the issue.

Reply:
Changed implementation to use `zipfile` instead of `shutil.make_archive`